File: 백준/Silver/4948. 베르트랑 공준/베르트랑 공준.js
const input = require('fs')
  .readFileSync(process.platform === 'linux' ? '/dev/stdin' : './input.txt')
  .toString()
  .trim()
  .split('\n')
  .map(Number);

function isPrime(n) {
  const array = new Array(2 * n + 1).fill(true);
  array[0] = array[1] = false;

  for (let i = 2; i * i <= 2 * n; i++) {
    if (array[i]) {
      for (let j = i * i; j <= 2 * n; j += i) {
        array[j] = false;
      }
    }
  }
  let result = 0;
  for (let i = n + 1; i <= 2 * n; i++) {
    if (array[i]) result++;
  }

  return result;
}

for (let i = 0; i < input.length - 1; i++) {
  console.log(isPrime(input[i]));
}

File: 백준/Silver/4948. 베르트랑 공준/베르트랑 공준.py
import sys
input = sys.stdin.read().splitlines()
array = list(map(int, input[:-1]))

MAX= 2 * max(array)

isPrime = [True] * (MAX + 1 )
isPrime[0] = isPrime[1] = False

for i in range (2, int(MAX**0.5)+1):
    if isPrime:
        for j in range(i*i, MAX+1, i):
            isPrime[j] = False

for n in array:
    count = sum(1 for i in range(n+1, 2*n+1) if isPrime[i])
    print(count)